hooks/useFetch.js

```diff
@@ -108,18 +104,27 @@ const useFetch = (url, refresh = false) => {
   const [loading, setLoading] = useState(true)
 
   useEffect(() => {
-    const { loading, data, refresh: startRefresh } = fetchData(url)
-    const sub = loading.subscribe(setLoading).add(
-      data.subscribe((ev) => {
-        setData(ev.value.body)
-        setError(ev.error)
+    const fetcher = fetchData(url).pipe(
+      tap((ev) => {
+        // maybe check for no errors? shrug
+        if (ev.hasValue) {
+          setLoading(false)
+        }
       }),
     )
 
-    if (refresh) {
-      sub.add(startRefresh.subscribe())
+    let refresher = fetcher
+    if (!refresh) {
+      refresher = fetcher.pipe(take(1))
```

Comment:
So, instead of taking the whole stream, this'll take a single one and then stop listening again? How does it fit in with the cache? Is the original stream kept in memory and the first item is taken? Or is it the latest item? Or am I not understanding at all? =)

Reply:
`take(1)` will emit a `complete` to the subscriber signaling that it's complete. If you don't just want avoid duplication of `fetcher` using a cache you should be able to use `publishBehavior` instead of `publish` to relay the last value emitted to subsequent subscribers. Let me add that.